### PR TITLE
Forecasting User Guide

### PR DESCRIPTION
**Pull request by Lovkush-A**
#### Reference Issues/PRs
See #361 

#### What does this implement/fix? Explain your changes.
First attempt at template of forecasting page, and of introduction

#### What should a reviewer concentrate their feedback on?

- [ ] Is the suggested template sensible?
- [ ] How do I get a bulleted-list without creating a new paragraph

#### Any other comments?

It is unclear to me what the difference is between the user guide and the tutorial. The tutorial seems to achieve the goals intended for the user guide.

##### For all contributions

- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/master/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/master/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

**Comment by mloning** [comment]
Thanks @Lovkush-A for working on this, take a look at the CONTRIBUTING.md on how to run the code quality checks automatically on your local machine! 

Ping me once you think this is in a good shape and ready for a review or if you get stuck!

**Comment by Lovkush-A** [comment]
Thanks @mloning for the tip. I have now got pre-commit installed (pretty neat!) and recommited.

Yes, I would like you to review, but in the informal sense. What I have done is not ready to be pulled. Questions I have:

1) Does it look like I have understood the intention behind the user guide?
2) What is the difference between the tutorial notebook and the user guide? From what I can tell, the tutorial already does most of what the user guide will do.

File: docs/source/user_guide/forecasting.rst [hunk 1]
```diff
@@ -9,3 +9,56 @@ Forecasting
...
+
+Examples.
+
+* Forecasting the global population
+* Forecasting the price of a stock
+* Forecasting the daily maximum temperature in a given location
+
+Example
+-------
+(Example about flights from tutorial)
```
Comment (mloning) [inline docs/source/user_guide/forecasting.rst:34]: please add a link to the actual examples

File: docs/source/user_guide/forecasting.rst [hunk 2]
```diff
@@ -9,3 +9,56 @@ Forecasting
...
+
+
+Algorithms included in sktime
+-----------------------------
+* Naive and seasonal naive
```
Comment (mloning) [inline docs/source/user_guide/forecasting.rst:39]: perhaps better to link to the API reference here: https://www.sktime.org/en/latest/api_reference.html#sktime-forecasting-time-series-forecasting

File: docs/source/user_guide/forecasting.rst [hunk 3]
```diff
@@ -9,3 +9,56 @@ Forecasting
...
+Evaluation and model selection
+------------------------------
+
+
+Algorithms not included in sktime
+---------------------------------
+
+
+Further reading
+_______________
```
Comment (mloning) [inline docs/source/user_guide/forecasting.rst:64]:
```suggestion
---------------
```

**Review by mloning — COMMENTED** [review]
As discussed @Lovkush-A - just a few minor suggestions. We could later copy more info from the forecasting tutorial notebook (references etc)

**Review by mloning — APPROVED** [review]
Thanks @Lovkush-A - looks all good to me now!

**Comment by mloning** [comment]
Now merging - thanks again @Lovkush-A!

**Pull request merged**